src/reducers/searchReducer.js

```diff
@@ -9,6 +9,8 @@ const initialState = {
 
 export default function Search(state = initialState, action) {
   switch (action.type) {
+    case 'SET_TOOL':
+      return (action.tool == 'search') ? state : initialState;
```

Comment:
ternarny operator je asi jedine miesto, kde pouzivam zatvorky aj ked nemusim, lebo to podla mna vyrazne zlepsuje jeho citatelnost. 
a `==` vs `===` -- ja som sa nikdy nestretol ze by mi to sposobilo problem, takze som z praxe vyhodnotil ze je strata casu sa tym zaoberat.